src/Ldap.php

```diff
@@ -165,7 +177,11 @@ public function getLastError(&$errorCode = null, array &$errorMessages = null)
          * can and eliminate dupes.
          */
         ErrorHandler::start(E_WARNING);
-        $estr1 = ldap_error($this->resource);
+        if (is_resource($this->resource)) {
+            $estr1 = ldap_error($this->resource);
+        } else {
```

Comment:
Can we init the `$estr1` here with `getLastError: could not call ldap_error ...` and reassign the result from `ldap_error` in case we have a proper resource? That would eliminate the `else`.